src/environmentd/src/bin/environmentd/main.rs

```diff
@@ -636,21 +637,7 @@ fn run(mut args: Args) -> Result<(), anyhow::Error> {
     {
         cors::Any.into()
     } else if !args.cors_allowed_origin.is_empty() {
-        let allowed = args.cors_allowed_origin.clone();
-        AllowOrigin::predicate(move |origin: &HeaderValue, _request_parts: _| {
-            for val in &allowed {
-                // For each specified allow cors origin, check if it starts with
-                // a *. and allow anything from that glob. Otherwise check for
-                // an exact match.
-                if (val.as_bytes().starts_with(b"*.")
-                    && origin.as_bytes().ends_with(&val.as_bytes()[1..]))
-                    || origin == val
-                {
-                    return true;
-                }
-            }
-            false
-        })
+        CorsChecker::new(args.cors_allowed_origin.iter()).into()
     } else {
         let port = args.http_listen_addr.port();
         AllowOrigin::list([
```

Comment:
It would be ideal to structure this logic as:

```
let allowed_origins = args.cors_allowed_origin.unwrap_or(&[HeaderValue::from_str("...")])
let cors_checker = CorsChecker::new(allowed_origins);
```

That will require pushing the logic for bare wildcards (`*`) into the `CorsChecker`.